ftw/contacts/utils.py

```diff
@@ -0,0 +1,199 @@
+from Products.CMFPlone.interfaces.siteroot import IPloneSiteRoot
+from Acquisition import aq_inner
+from Acquisition import aq_parent
+from base64 import b64encode
+from plone import api
+from plone.memoize import instance
+from plone.namedfile.utils import stream_data
+from StringIO import StringIO
+from zc.relation.interfaces import ICatalog
+from zope.component import getUtility
+from zope.intid.interfaces import IIntIds
+import unicodedata
+
+
+ALPHABET = map(chr, range(ord('A'), ord('Z') + 1))
+LETTERS = ALPHABET + ['#']
+
+
+def encode(text):
+    if not text:
+        return ''
+    if isinstance(text, unicode):
+        return text.encode('utf-8')
+    return text
+
+
+def get_organization(context, portal_types=[]):
+    """Traverse up to search a parent-object matching a portal_type
+    If it reaches a portal_type, it returns its title.
+    If it reaches the plone siteroot it returns an empty string
+    """
+    context = aq_inner(context)
+    parent = aq_parent(context)
+    while parent.portal_type not in portal_types:
+        if IPloneSiteRoot.providedBy(parent):
+            return ''
+        parent = parent.aq_parent
+    return parent.Title()
+
+
+def safe_html(text):
+    if not text:
+        return ''
+
+    if isinstance(text, unicode):
+        text = text.encode('utf-8')
+
+    ttool = api.portal.get_tool('portal_transforms')
+    text = ttool.convertTo('text/x-html-safe', text).getData()
+
+    # we need html br
+    text = text.replace('\r\n', '<br />').replace('\n', '<br />')
```

Comment:
- `br`s / newline: i'd do it like your example and not replace newlines manually.
- `p`s: why do you want to remove `p`s in the first place? Is it because of padding / margin? How about styling it with CSS and keeping `p`s?